python/pyspark/sql/connect/client.py

```diff
@@ -628,21 +612,33 @@ def _handle_error(self, rpc_error: grpc.RpcError) -> NoReturn:
                 if d.Is(error_details_pb2.ErrorInfo.DESCRIPTOR):
                     info = error_details_pb2.ErrorInfo()
                     d.Unpack(info)
-                    if info.reason == "org.apache.spark.sql.AnalysisException":
+                    reason = info.reason
+                    if reason == "org.apache.spark.sql.AnalysisException":
                         raise SparkConnectAnalysisException(
-                            info.reason, info.metadata["message"], info.metadata["plan"]
```

Comment:
I don't use `reason` from `SparkConnectAnalysisException` and other specified Exceptions (e.g. `SparkConnectParseException`, `SparkConnectTempTableAlreadyExistsException`), since I believe the name of exception sounds self-explanatory.

Reply:
sounds good, I just check the scala side also throw `org.apache.spark.sql.catalyst.analysis.TempTableAlreadyExistsException` which inherit `AnalysisException`